apollo-idling-resource/src/main/java/com/apollographql/apollo3/android/ApolloIdlingResource.kt

```diff
@@ -38,13 +45,20 @@ class ApolloIdlingResource(
 }
 
 fun ApolloClient.Builder.idlingResource(idlingResource: ApolloIdlingResource): ApolloClient.Builder {
-  return addFlowDecorator {
-    it.onStart {
-      idlingResource.operationStart()
-    }.onCompletion {
-      idlingResource.operationEnd()
+  return addInterceptor(object : ApolloInterceptor {
```

Comment:
Ideally, it could be nice to disallow calling `idlingResource` several times on the same builder - I don't see an obvious way to implement it. May be over the top anyway :)

Reply:
We could make `ApolloClient.Builder.interceptors` public and check for an instance of a previously registered interceptor. That's a tiny performance hit but certainly negligible (and only used in tests anyways)

Reply:
Implemented it [here](https://github.com/apollographql/apollo-android/pull/3546) - not 100% convinced, so feel free to close if you think it's too much 😄

Reply:
Merged!